### PR TITLE
Storage refactoring (SQL statements as static strings)

### LINKED ISSUE #390: Make SQL statement as static strings
SQL statements do not need to be build every time, we should have them as static strings.

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes #390 

# Test Plan:
Usual tests

### CODE REVIEW

File: consensus/tangle/tangle.c [hunk 1]
```diff
@@ -101,6 +85,20 @@ retcode_t iota_tangle_transaction_load_hashes_of_approvers(
...
+    const tangle_t *const tangle, flex_trit_t *const hash,
+    uint64_t snapshot_index) {
+  return iota_stor_transaction_update_snapshot_index(&tangle->conn, hash,
+                                                     snapshot_index);
+}
+
+retcode_t iota_tangle_transaction_exist(const tangle_t *const tangle,
+                                        transaction_field_t const field,
+                                        const trit_array_p key, bool *exist) {
+  // TODO - when more "persistance provider are implemented - should loop
```
Comment (th0br0) [inline consensus/tangle/tangle.c]: Can you remove this comment? There's arguably only ever one persistance provider per build.

File: common/storage/sql/statements.c [hunk 2]
```diff
@@ -5,236 +5,75 @@
...
+char *iota_statement_transaction_update_snapshot_index =
+    "UPDATE " TRANSACTION_TABLE_NAME " SET " TRANSACTION_COL_SNAPSHOT_INDEX
+    "=? WHERE " TRANSACTION_COL_HASH "=?";
+
+char *iota_statement_transaction_exist =
+    "SELECT '1' WHERE EXISTS(SELECT 1 "
+    "FROM " TRANSACTION_TABLE_NAME ")";
+
+char *iota_statement_transaction_exist_by_hash =
+    "SELECT '1' WHERE EXISTS(SELECT 1 "
```
Comment (th0br0) [inline common/storage/sql/statements.c]: This should probably be `SELECT 1 WHERE EXISTS`?

File: common/storage/sql/statements.c [hunk 3]
```diff
@@ -5,236 +5,75 @@
...
+char *iota_statement_milestone_select_last =
+    "SELECT * FROM " MILESTONE_TABLE_NAME " ORDER BY " MILESTONE_COL_INDEX
+    " DESC LIMIT 1";
+
+char *iota_statement_milestone_select_next =
+    "SELECT * FROM " MILESTONE_TABLE_NAME " WHERE " MILESTONE_COL_INDEX
+    ">? ORDER BY " MILESTONE_COL_INDEX " ASC LIMIT 1";
+
+char *iota_statement_milestone_exist =
+    "SELECT '1' WHERE EXISTS(SELECT 1 "
```
Comment (th0br0) [inline common/storage/sql/statements.c]: same here, why the char 1? :thinking:

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/storage/sql/statements.c (hunk 3 above)
Comment (thibault-martinez) [reply common/storage/sql/statements.c]: No idea, I'm modifying it; unless @tsvisabo has some more input on this ?

**Review by th0br0 — APPROVED** [review]

**Pull request merged**